source/comtypesMonkeyPatches.py

```diff
@@ -95,14 +96,21 @@ def new__call__(self,*args,**kwargs):
 # This causes Release() to be called more than it should, which is very nasty and will eventually cause us to access pointers which have been freed.
 from comtypes import _compointer_base
 
-_cpbDel = _compointer_base.__del__
+_compointer_base._oldCpbDel = _compointer_base.__del__
 def newCpbDel(self):
+	# __del__ may be called while Python is exiting.
+	# In this state, global symbols may be set to None
+	# Therefore avoid calling into garbageHandler.
+	# Using local variables or calling other methods on this class is still okay.
+	isFinalizing = getattr(sys, 'is_finalizing', lambda: True)()

```

Comment:
This seems wrong, but it probably works by accident, a function type is truthy, but isn't actually being called.
```suggestion
	isFinalizing = getattr(sys, 'is_finalizing', True)()

```

Reply:
I did not review this whole PR. But @feerrenrut, your suggestion does not seem correct to me. @michaelDCurran's initial code seems correct.
Did you notice the parenthesis pair at the end of the line?
If is_finalizing is not an attribute of sys, True will be called as a function, what does not seem intended.

Reply:
Note that:
* sys.is_finalizer is a function.
*  sys is a module, except for on exit it may be None.
* ```is_finalizer = getattr(sys, 'is_finalizer', lambda: True)()``` works as follows:
 * if sys is the valid sys module, getattr will return the is_finalizer function
 * If sys is None, getattr will return a lambda that if executed will return True
 * We then execute the result from getattr (executing either the real is_finalizer function, or the lambda).
Note the reason why we can't just check if sys is None by itself is that it may become None after that check. Thus, we use getattr as it is atomic.

Reply:
If you confirm it is atomic, maybe you could use a try ... except AtributeError ... statement instead. This would be more pythonic and self-explaining IMO. Just a suggestion.

Reply:
> Did you notice the parenthesis pair at the end of the line?

My mistake, I missed the parenthesis. I didn't realize that `sys.is_finalizer` was a function. I think this could be made easier to read. Eg as @CyrilleB79 suggests in a try block, or like:
```python
isFinalizingFunc = getattr(sys, 'is_finalizing', lambda: True)
isFinalizing = isFinalizingFunc()
```